File: backend/src/main/java/com/apocalypse/thefall/entity/item/Armor.java
package com.apocalypse.thefall.entity.item;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@Table(name = "armor")
@DiscriminatorValue("ARMOR")
@SuperBuilder
@NoArgsConstructor
public class Armor extends Item {

    @Column(name = "armor_class", nullable = false)
    private Integer armorClass;

    @JsonManagedReference
    @OneToMany(mappedBy = "armor", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private Set<ArmorDamage> damages = new HashSet<>();
}

File: backend/src/main/java/com/apocalypse/thefall/entity/item/Armor.kt
package com.apocalypse.thefall.entity.item

import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*
import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import org.hibernate.annotations.Fetch
import org.hibernate.annotations.FetchMode

@Entity
@Table(name = "armor")
@DiscriminatorValue("ARMOR")
open class Armor : Item() {

    @Column(name = "armor_class", nullable = false)
    open var armorClass: Int = 0

    @Cache(usage = CacheConcurrencyStrategy.READ_ONLY)
    @Fetch(FetchMode.SUBSELECT)
    @JsonManagedReference
    @OneToMany(mappedBy = "armor", cascade = [CascadeType.ALL], orphanRemoval = true)
    open var damages: MutableSet<ArmorDamage> = mutableSetOf()
}
